com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/providers/SettingsProvider.java

```diff
@@ -14,23 +14,91 @@
 
 package com.ca.lsp.cobol.service.providers;
 
+import com.broadcom.lsp.domain.cobol.databus.api.DataBusBroker;
+import com.broadcom.lsp.domain.cobol.event.api.EventObserver;
+import com.broadcom.lsp.domain.cobol.event.model.DataEventType;
+import com.broadcom.lsp.domain.cobol.event.model.FetchedSettingsEvent;
 import com.ca.lsp.cobol.model.ConfigurationSettingsStorable;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
 import com.google.inject.Provider;
 import com.google.inject.Singleton;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
+import java.lang.reflect.Type;
 
+/**
+ * This class is resposible to keep the settings configuration provided by the user in the setting
+ * storage system and that will be consumed by the {@link
+ * com.ca.lsp.cobol.service.CopybookServiceImpl}
+ */
 @Singleton
-public class SettingsProvider implements Provider<ConfigurationSettingsStorable> {
+@Slf4j
+public class SettingsProvider
+    implements Provider<ConfigurationSettingsStorable>, EventObserver<FetchedSettingsEvent> {
   private ConfigurationSettingsStorable configurationSettingsStorable;
+  @Getter private DataBusBroker databus;
 
-  public void set(@Nonnull ConfigurationSettingsStorable configurationSettingsStorable) {
+  public SettingsProvider() {}
+
+  @Builder
+  public SettingsProvider(DataBusBroker databus) {
+    this.databus = databus;
+    this.databus.subscribe(DataEventType.FETCHED_SETTINGS_EVENT, this);
+  }
+
+  public void set(@Nullable ConfigurationSettingsStorable configurationSettingsStorable) {
     this.configurationSettingsStorable = configurationSettingsStorable;
   }
 
+  /**
+   * The method invoked by a provider return the object that represent the user settings config.
+   *
+   * @return a {@link ConfigurationSettingsStorable} fullfilled with the user settings or an empty
+   *     one if not present
+   */
   @Override
   @Nullable
   public ConfigurationSettingsStorable get() {
-    return this.configurationSettingsStorable;
+
+    if (configurationSettingsStorable != null) {
+      return deepCopy(configurationSettingsStorable);
+    } else {
+      return ConfigurationSettingsStorable.builder().build();
+    }
+  }
+
+  /** @param databus the channel used for receive message about {@link FetchedSettingsEvent} */
+  @Inject
+  public void setDatabus(@Nonnull DataBusBroker databus) {
+    this.databus = databus;
+    this.databus.subscribe(DataEventType.FETCHED_SETTINGS_EVENT, this);
+  }
+
+  private static ConfigurationSettingsStorable deepCopy(
```

Comment:
Technically, it is not deep copying due to you build a new instance using the same objects